controller/lib/hal/hal_stm32.cpp

```diff
@@ -784,16 +769,15 @@ static void EnableClock(void *ptr) {
       //      {TIMER16_BASE, 6, 17},
   };
 
-  // I don't include all the peripherials here, just the ones
-  // that we currently use or seem likely to be used in the
-  // future.  To add more peripherials, just look up the appropriate
-  // bit in the reference manual RCC chapter.
+  // I don't include all the peripherials here, just the ones that we currently
+  // use or seem likely to be used in the future.  To add more peripherials,
+  // just look up the appropriate bit in the reference manual RCC chapter.
   int ndx = -1;
   int bit = 0;
-  for (uint32_t i = 0; i < sizeof(rccInfo) / sizeof(rccInfo[0]); i++) {
-    if (ptr == rccInfo[i].base) {
-      ndx = rccInfo[i].ndx;
-      bit = rccInfo[i].bit;
+  for (auto &info : rccInfo) {
+    if (ptr == info.base) {
+      ndx = info.ndx;
+      bit = info.bit;
```

Comment:
More interesting foreign syntax.  I really need to get with the times!

Reply:
This one the compiler was even able to suggest and auto-translate to!  Amazing times we live in.  :)